tcms_django_plugin/result.py

```diff
@@ -116,11 +116,22 @@ def addError(self, test, err):
         super().addError(test, err)
         self.debug_sql_stream.seek(0)
         self.errors[-1] = self.errors[-1] + (self.debug_sql_stream.read(),)
+        self.backend.add_comment(self.test_execution_id, self.errors[-1][2])
 
     def addFailure(self, test, err):
         super().addFailure(test, err)
         self.debug_sql_stream.seek(0)
         self.failures[-1] = self.failures[-1] + (self.debug_sql_stream.read(),)
+        self.backend.add_comment(self.test_execution_id, self.failures[-1][2])
+
+    def addSubTest(self, test, subtest, err):
+        super().addSubTest(test, subtest, err)
+        if err is not None:
+            self.debug_sql_stream.seek(0)
+            errors = self.failures if issubclass(
+                err[0], test.failureException) else self.errors
+            errors[-1] = errors[-1] + (self.debug_sql_stream.read(),)
+            self.backend.add_comment(self.test_execution_id, errors[-1][2])
```

Comment:
Everything except the last line has been copied from the Django sources and I don't like this. Not to mention when Django changes we'll get out of sync.

Reply:
I get the problem here - on one hand you need to mimic whatever Django is doing to capture the SQL debug statements OTOH you need to send all of this into Kiwi TCMS in the same way that the TestResult class already does.

I am thinking in the direction of multiple inheritance, MRO and very likely mixins. It looks to me that the most elegant and flexible way to deal with this will be to have the Kiwi TCMS specifics into a mixin class and then the actual test result classes will inherit from both Django and the mixin and glue the two together.  This is not a simple change so it can be left for a separate pull request.

However with such an implementation I'd still want to void having to override every single method and call super() + a mixin class method to send the results to the Kiwi TCMS DB because we'll end up with 2 classes which have identical methods calling super() + something else. Ideally the actual interaction with the API will be kept in a single place but I don't have ideas ATM. This will need some experimentation.

Reply:
So what do we do for now? Not sure what to do next in regard to this.

Reply:
Leave it like this for now. Resolve the rest and I will merge this PR as-is. I will experiment with this code a little bit to figure out if there's something else that can be done here.